src/core/server/core_app/bundle_routes/utils.ts

```diff
@@ -6,33 +6,19 @@
  * Side Public License, v 1.
  */
 
+import { createReadStream, Stats } from 'fs';
 import { createHash } from 'crypto';
-import Fs from 'fs';
-
 import * as Rx from 'rxjs';
-import { takeUntil, map } from 'rxjs/operators';
-
-import { FileHashCache } from './file_hash_cache';
-
-/**
- *  Get the hash of a file via a file descriptor
- */
-export async function getFileHash(cache: FileHashCache, path: string, stat: Fs.Stats, fd: number) {
-  const key = `${path}:${stat.ino}:${stat.size}:${stat.mtime.getTime()}`;
-
-  const cached = cache.get(key);
-  if (cached) {
-    return await cached;
-  }
+import { map, takeUntil } from 'rxjs/operators';
 
+export const generateFileHash = (fd: number): Promise<string> => {
   const hash = createHash('sha1');
-  const read = Fs.createReadStream(null as any, {
+  const read = createReadStream(null as any, {
     fd,
     start: 0,
     autoClose: false,
   });
-
-  const promise = Rx.merge(
+  return Rx.merge(
     Rx.fromEvent<Buffer>(read, 'data'),
     Rx.fromEvent<Error>(read, 'error').pipe(
       map((error) => {
```

Comment:
Not sure of the value of using an observable approach to generate the hash tbh, but it's working, so I kept the implementation as it was in legacy.

Reply:
Yeah I'm sure this creates 100s of RxJs objects, but guess fine for now.